src/logic/user.ts

```diff
@@ -0,0 +1,94 @@
+import express from "express";
+import models from "../models/index";
+import Post from "../logic/post";
+
+const postMethods = new Post;
+
+export default class User {
+
+  async showUsers(req: express.Request, res: express.Response) {
+    const result = await models.user.find();
+    res.send(result);
+  }
+
+  async createUser(req: express.Request, res: express.Response) {
+    const result = await models.user.create({
+      username: req.body.username,
+      password: req.body.password,
+      gender: req.body.gender,
+      isAuthenticated: false
+    });
+    res.send(result);
+  }
+  // CHEACK deleteAllPostOfUser
+  async deleteUser(req: express.Request, res: express.Response) {
+    const result = await models.user.findOne({ username: req.body.username });
+    if (result) {
+      if (result.isAuthenticated == true) {
```

Comment:
It's looks bad. Rewrite this. Avoid using a lot of **if/else** and ```especially``` nested **if/else**